processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -469,7 +469,8 @@ public boolean isAssignableTo(Type other) {
             // the current target accessor can also be a getter method.
             // The following if block, checks if the target accessor should be overruled by an add method.
             if ( cmStrategy == CollectionMappingStrategyPrism.SETTER_PREFERRED
-                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED ) {
+                || cmStrategy == CollectionMappingStrategyPrism.ADDER_PREFERRED
+                || cmStrategy == CollectionMappingStrategyPrism.TARGET_IMMUTABLE ) {
```

Comment:
> Thinking a bit about this. ADDER_PREFERRED does not mean list.add(something), it means entity.add(something), right?

That's true.. is that not clear from the doc?


`Only target add- Available` .. This cannot be. An adder can only be there when there is also an getter and/or setter..  So I think the table is wrong to start with.. So I think the column should disappear al together.  I'll do that in a separate PR.. Want to close this one.